pkg/cloudfoundry/ManifestUtils.go

```diff
@@ -0,0 +1,226 @@
+package cloudfoundry
+
+import (
+	"fmt"
+	"github.com/ghodss/yaml"
+	"github.com/pkg/errors"
+	"io/ioutil"
+	"reflect"
+
+	"github.com/SAP/jenkins-library/pkg/log"
+)
+
+const propApplications = "applications"
+const propBuildpacks = "buildpacks"
+const propBuildpack = "buildpack"
+
+// Manifest ...
+type Manifest struct {
+	self     map[string]interface{}
+	modified bool
+	name     string
+}
+
+var m Manifest
+
+var _readFile = ioutil.ReadFile
+
+// ReadManifest ...
+func ReadManifest(name string) (Manifest, error) {
+
+	log.Entry().Infof("Reading manifest file  '%s'", name)
+
+	m := Manifest{self: make(map[string]interface{}), name: name, modified: false}
+
+	content, err := _readFile(name)
+	if err != nil {
+		return m, errors.Wrapf(err, "cannot read file '%v'", m.name)
+	}
+
+	err = yaml.Unmarshal(content, &m.self)
+
+	if err != nil {
+		return m, errors.Wrapf(err, "Cannot parse yaml file '%s': %s", m.name, string(content))
+	}
+
+	log.Entry().Infof("Manifest file '%s' has been parsed", m.name)
+
+	return m, nil
+}
+
+// WriteManifest ...
+func (m *Manifest) WriteManifest() error {
+
+	d, err := yaml.Marshal(&m.self)
+	if err != nil {
+		return err
+	}
+
+	log.Entry().Debugf("Writing manifest file '%s'", m.name)
+	err = ioutil.WriteFile(m.name, d, 0644)
+
+	if err == nil {
+		m.modified = false
+	}
+
+	log.Entry().Debugf("Manifest file '%s' has been written", m.name)
+	return err
+}
+
+// GetName ...
+func (m Manifest) GetName() string {
+	return m.name
+}
+
+// GetApplications ...
+func (m Manifest) GetApplications() ([]interface{}, error) {
+	return toSlice(m.self)
+}
+
+// ApplicationHasProperty Checks if the application denoted by 'index' has the property 'name'
+func (m Manifest) ApplicationHasProperty(index int, name string) (bool, error) {
+
+	sliced, err := toSlice(m.self[propApplications])
+
+	if err != nil {
+		return false, err
+	}
+
+	if index >= len(sliced) {
+		return false, fmt.Errorf("Index (%d) out of bound. Number of apps: %d", index, len(sliced))
+	}
+
+	_m, err := toMap(sliced[index])
+
+	if err != nil {
+		return false, err
+	}
+
+	_, ok := _m[name]
+
+	return ok, nil
+}
+
+// GetApplicationProperty ...
+func (m Manifest) GetApplicationProperty(index int, name string) (interface{}, error) {
+
+	sliced, err := toSlice(m.self[propApplications])
+
```

Comment:
Remove empty line.

Reply:
What is wrong with the empty lines. Basically I don't like fighting for/against empty lines. I would better understand if there are several consecutive empty lines. But here we have only single empty lines. Do we have some coding standards (not covered by formatters) handling with empty lines?

Reply:
I guess the error check belongs to the assignment, that's why the empty line looks a bit out of place.

Reply:
The point is that the next thing in the code is the check on the error, so it fits to keep this close together. I[n other places you have done it like that, that the error check immediately follows the retrieval of it](https://github.com/SAP/jenkins-library/blob/09c9ef5d43981678c7f49e6fbe76104a9032b463/pkg/cloudfoundry/ManifestUtils.go#L152). I would suggest you keep it one way or the other. But of course if you say the effort is not necessary then that is up to you.

Reply:
I feel I'm not made for discussions like that. Maybe my in-brain formatter strips the empty lines anyway. However - I removed all blank lines dealing with checking for errors.